srml/eth-relay/src/lib.rs

```diff
@@ -201,7 +210,11 @@ impl<T: Trait> Module<T> {
 	/// 2. proof of pow (mixhash)
 	/// 3. challenge
 	fn verify_header(header: &EthHeader) -> Result {
-		// TODO: check parent hash,
+		ensure!(
+			header.hash() == header.re_compute_hash(),
+			"Header Hash Check Failed - Hash NOT MATCH",
```

Comment:
I think `Header Hash - MISMATCHED` is better.

Could you please replace all `NOT MATCHED` with `MISMATCHED`?